### PR TITLE
COM-12716: Fix AV1 argon test vector paths when running tests

### LINKED ISSUE #218: Argon AV1 test resource patch don't match the downloaded resources path
I wanted to try the new Argon suite, but its broken, the resources and the path don't match. Using GStreamer VA AV1, I'm getting this comment:

```
gst-launch-1.0 --no-fault filesrc location=/home/nicolas/Sources/fluster/fluster/../resources/AV1_ARGON_VECTORS/test3534_116/argon_coveragetool_av1_base_and_extended_profiles_v2.1/profile0_core/streams/test3534_116.obu ! parsebin ! vaav1dec ! video/x-raw ! videoconvert dither=none ! video/x-raw,format=I420_10LE ! videocodectestsink -m
```

But the path is actually:
```
resources/AV1_ARGON_VECTORS/argon_coveragetool_av1_base_and_extended_profiles_v2.1/profile0_core/streams/test3534_116.obu
```

### PR DESCRIPTION
**Pull request by mdimopoulos**
- added new is_single_archive optional member in TestSuite class
- excluded this member from json export function
- adapted AV1 argon test suite generator script and .json

fixes #218

### CODE REVIEW

**Comment by ylatuya** [comment]
@mdimopoulos the decription should have any of the link [keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link this PR to the [issue](https://github.com/fluendo/fluster/issues/218) it fixes

File: fluster/test.py [hunk 1]
```diff
@@ -69,7 +69,11 @@ def _test(self) -> None:
         input_filepath = os.path.join(
             self.resources_dir,
             self.test_suite.name,
-            self.test_vector.name,
+            (
+                self.test_vector.name
+                if self.test_suite.is_single_archive is False
```
Comment (rgonzalezfluendo) [inline fluster/test.py:74]:
```suggestion
                if not self.test_suite.is_single_archive
```

File: fluster/test_suite.py [hunk 2]
```diff
@@ -176,6 +178,8 @@ def to_json_file(self, filename: str) -> None:
             data.pop("filename")
             data.pop("test_vectors_success")
             data.pop("time_taken")
+            if self.is_single_archive is False:
```
Comment (rgonzalezfluendo) [inline fluster/test_suite.py:181]:
```suggestion
            if not self.is_single_archive:
```

File: fluster/test_suite.py [hunk 3]
```diff
@@ -346,7 +350,7 @@ def _callback_error(err: Any) -> None:
 
             downloads = []
 
-            if self.name != "AV1_ARGON_VECTORS":
+            if self.is_single_archive is False:
```
Comment (rgonzalezfluendo) [inline fluster/test_suite.py:353]:
```suggestion
            if not self.is_single_archive:
```

File: test_suites/av1/AV1_ARGON_VECTORS.json [hunk 4]
```diff
@@ -2,6 +2,7 @@
     "name": "AV1_ARGON_VECTORS",
     "codec": "AV1",
     "description": "AV1 Argon Streams",
+    "is_single_archive": true,
```
Comment (rgonzalezfluendo) [inline test_suites/av1/AV1_ARGON_VECTORS.json:5]: :heart:

File: fluster/test.py (hunk 1 above)
Comment (mdimopoulos) [reply fluster/test.py]: Black doesn't like this. Maybe when we merge the ruff PR this will be allowed

**Review by rgonzalezfluendo — APPROVED** [review]

**Comment by ndufresne** [comment]
Thanks for working on that, meanwhile we started working on GStreamer (typefinder and parser) to support raw obu.

**Pull request merged**